modules/ui/rapid_view_manage_datasets.js

```diff
@@ -216,30 +228,88 @@ export function uiRapidViewManageDatasets(context, parentModal) {
 
 
   function renderDatasets(selection) {
+    const status = selection.selectAll('.rapid-view-manage-datasets-status');
+    const results = selection.selectAll('.rapid-view-manage-datasets');
+
     const showPreview = prefs('rapid-internal-feature.previewDatasets') === 'true';
     const service = services.esriData;
+
     if (!service || (Array.isArray(_datasetInfo) && !_datasetInfo.length)) {
-      selection.text(t('rapid_feature_toggle.esri.no_datasets'));
+      results.classed('hide', true);
+      status.classed('hide', false).text(t('rapid_feature_toggle.esri.no_datasets'));
       return;
     }
 
     if (!_datasetInfo) {
-      selection.text(t('rapid_feature_toggle.esri.fetching_datasets'));
+      results.classed('hide', true);
+      status.classed('hide', false)
+        .text(t('rapid_feature_toggle.esri.fetching_datasets'));
+
+      status
+        .append('br');
+
+      status
+        .append('img')
+        .attr('class', 'rapid-view-manage-datasets-spinner')
+        .attr('src', context.imagePath('loader-black.gif'));
+
       service.loadDatasets()
         .then(results => {
-          // exclude preview datasets unless user has opted into them
-          return _datasetInfo = Object.values(results)
-            .filter(d => showPreview || !d.groupCategories.some(category => category === '/Categories/Preview'));
+          // Build set of available categories
+          let categories = new Set();
+
+          Object.values(results).forEach(d => {
+            d.groupCategories.forEach(c => {
+              categories.add(c.toLowerCase().replace('/categories/', ''));
```

Comment:
Ah, this is swank. I see you're constructing the categories on-the-fly already, this is excellent!